pkg/api/getambassador.io/v3alpha1/crd_mapping.go

```diff
@@ -79,6 +79,9 @@ type MappingSpec struct {
 	Timeout     *MillisecondDuration `json:"timeout_ms,omitempty"`
 	IdleTimeout *MillisecondDuration `json:"idle_timeout_ms,omitempty"`
 	TLS         string               `json:"tls,omitempty"`
+	// +kubebuilder:validation:MinItems=1
+	HealthChecks    []HealthCheck `json:"health_checks,omitempty"`
+	HealthCheckPort *int          `json:"health_check_port,omitempty"`
```

Comment:
I removed health_check_port since Envoy only has config to support it when specifying Endpoints manually. Unfortunately the endpoint resolver does not support this configuration despite the fact that Envoy says that Health checking is meant to be used with the endpoint resolver for the most benefit.